src/codegen.jl

```diff
@@ -131,118 +124,88 @@ end
 
 function build_bitmask(df::Any, condition::Any)::Expr
     condition = condition isa Nothing ? true : condition
-    mask = replace_variable_references(df, condition) |> vectorize_function_calls
+    mask = replace_column_references(df, condition) |> vectorize_function_calls
     :(falses(nrow($(df))) .| Missings.replace($mask, false))
 end
 
 build_bitmask(command::Command) = isnothing(command.condition) ? :(trues(nrow($(command.df)))) : build_bitmask(command.df, command.condition)
 
 function split_assignment(expr::Any)
-    if isassignment(expr)
-        return (expr.args[1], expr.args[2])
-    else
-        ArgumentError("Expected assignment expression, got $expr") |> throw
-    end
+    isassignment(expr) || (ArgumentError("Expected assignment expression, got $expr") |> throw)
+    (expr.args[1], expr.args[2])
 end
 
-function extract_function_references(expr::Any)
-    if is_function_call(expr) || (expr isa Expr && expr.head == :call && is_operator(expr.args[1]))
-        return vcat([expr.args[1]], extract_function_references.(expr.args[2:end])...)
-    elseif expr isa Expr
-        return vcat(extract_function_references.(expr.args)...)
-    else
-        return Symbol[]
-    end
+extract_column_references(::Any) = Symbol[]
```

Comment:
here `expr` is missing from argument

Reply:
If the input is not used to evaluate the function, it is enough to note its type for dispatch.

Reply:
I see